lib/ensure_content_type.js

```diff
@@ -47,7 +47,13 @@ function checkContentType(hyper, req, next, expectedContentType, responsePromise
             return res;
         }
 
-        if (res.headers && res.headers['content-type'] !== expectedContentType) {
+        if (res.headers && res.headers === undefined) {
```

Comment:
I fail to see how this will ever evaluate to `true`. If `res.headers === undefined` then `res.headers` will always yield a false outcome, which means that the outcome will always be `false`. I also fail to see the benefit of deleting the `content-type` key in this circumstance.

Reply:
Oh this is obviously a bug, it should read `res.headers['content-type'] === undefined`